In [55]:
import pandas as pd

In [56]:
df = pd.read_csv(r"CSVversions\Global Data Superstore.csv")
df.head()

Row ID         Order ID  Order Date   Ship Date       Ship Mode  \
0    1783   MX-2014-116267  31-12-2014  03-01-2015    Second Class   
1   26535    IN-2014-43550  31-12-2014  01-01-2015     First Class   
2   44025     RS-2014-1460  31-12-2014  02-01-2015    Second Class   
3   26333    IN-2014-30390  31-12-2014  03-01-2015     First Class   
4   12929  ES-2014-3458802  31-12-2014  05-01-2015  Standard Class   

  Customer ID        Customer Name    Segment        City          State  ...  \
0    EB-13975           Erica Bern  Corporate   S�o Paulo      S�o Paulo  ...   
1    ML-17395  Marina Lichtenstein  Corporate     Jakarta        Jakarta  ...   
2     PB-9105         Peter B�hler   Consumer         Ufa  Bashkortostan  ...   
3    JD-16150     Justin Deggeller  Corporate     Bangkok        Bangkok  ...   
4    JG-15805           John Grady  Corporate  Maidenhead        England  ...   

         Product ID    Category Sub-Category  \
0   TEC-CO-10000137  Technology      Copiers   
1   FUR-BO-10004679   Furniture    Bookcases   
2  TEC-HEW-10004652  Technology      Copiers   
3   FUR-TA-10001531   Furniture       Tables   
4   TEC-PH-10003683  Technology       Phones   

                              Product Name      Sales Quantity Discount  \
0                Canon Wireless Fax, Color  1264.4660        5    0.002   
1           Safco Library with Doors, Pine  1091.2806        3    0.070   
2           Hewlett Personal Copier, Laser   865.6200        6    0.000   
3  Lesro Training Table, Adjustable Height  1048.7313        9    0.570   
4                Motorola Audio Dock, VoIP   867.3000        5    0.000   

      Profit  Shipping Cost  Order Priority  
0   301.4660         253.25            High  
1    46.9206         243.11            High  
2    51.8400         138.18            High  
3 -1195.2387          86.86            High  
4   251.4000          53.16          Medium  

[5 rows x 24 columns]

DATA PREPROCESSING and CLEANING

In [57]:
df.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'City', 'State', 'Country',
       'Postal Code', 'Market', 'Region', 'Product ID', 'Category',
       'Sub-Category', 'Product Name', 'Sales', 'Quantity', 'Discount',
       'Profit', 'Shipping Cost', 'Order Priority'],
      dtype='object')

DELETE UNNECESSARY COLUMNS/FEATURES

In [58]:
# Row_ID, Order ID, Cutomer ID, Customer Name, Product ID are not required.

df = df[['Order Date', 'Ship Date', 'Ship Mode',
       'Segment', 'City', 'State', 'Country',
       'Postal Code', 'Market', 'Region', 'Category',
       'Sub-Category', 'Product Name', 'Sales', 'Quantity', 'Discount',
       'Profit', 'Shipping Cost', 'Order Priority']]
df.head()

Order Date   Ship Date       Ship Mode    Segment        City  \
0  31-12-2014  03-01-2015    Second Class  Corporate   S�o Paulo   
1  31-12-2014  01-01-2015     First Class  Corporate     Jakarta   
2  31-12-2014  02-01-2015    Second Class   Consumer         Ufa   
3  31-12-2014  03-01-2015     First Class  Corporate     Bangkok   
4  31-12-2014  05-01-2015  Standard Class  Corporate  Maidenhead   

           State         Country  Postal Code Market          Region  \
0      S�o Paulo          Brazil          NaN  LATAM           South   
1        Jakarta       Indonesia          NaN   APAC  Southeast Asia   
2  Bashkortostan          Russia          NaN   EMEA            EMEA   
3        Bangkok        Thailand          NaN   APAC  Southeast Asia   
4        England  United Kingdom          NaN     EU           North   

     Category Sub-Category                             Product Name  \
0  Technology      Copiers                Canon Wireless Fax, Color   
1   Furniture    Bookcases           Safco Library with Doors, Pine   
2  Technology      Copiers           Hewlett Personal Copier, Laser   
3   Furniture       Tables  Lesro Training Table, Adjustable Height   
4  Technology       Phones                Motorola Audio Dock, VoIP   

       Sales  Quantity  Discount     Profit  Shipping Cost Order Priority  
0  1264.4660         5     0.002   301.4660         253.25           High  
1  1091.2806         3     0.070    46.9206         243.11           High  
2   865.6200         6     0.000    51.8400         138.18           High  
3  1048.7313         9     0.570 -1195.2387          86.86           High  
4   867.3000         5     0.000   251.4000          53.16         Medium

CHECK FOR DUPLICATE ROWS

In [59]:
#check duplicate rows
print(df[df.duplicated(df.head(0))])
#NO DUPLICATE ROWS FOUND

Empty DataFrame
Columns: [Order Date, Ship Date, Ship Mode, Segment, City, State, Country, Postal Code, Market, Region, Category, Sub-Category, Product Name, Sales, Quantity, Discount, Profit, Shipping Cost, Order Priority]
Index: []


CHECK FOR NULL OR EMPTY VALUES

In [60]:
df['Postal Code'].isnull().sum()

41296

As we can see Postal Code has 41296 rows as NaN. it is an unecessary column. The coutry and city data is enough to define the location

In [61]:
del df['Postal Code']

CONVERT CATEGORICAL DATA TO INTEGER

In [62]:
newdf = df.copy(deep=True)

In [63]:
newdf['Ship Mode'].value_counts()

Standard Class    30774
Second Class      10309
First Class        7505
Same Day           2701
Name: Ship Mode, dtype: int64

In [64]:
newdf['Ship Mode'] =  newdf['Ship Mode'].map({'Same Day':0, 
                                              'First Class':1,
                                              'Second Class':2,
                                              'Standard Class':3})
newdf['Ship Mode'].value_counts()

3    30774
2    10309
1     7505
0     2701
Name: Ship Mode, dtype: int64

In [65]:
newdf['Segment'].value_counts()

Consumer       26518
Corporate      15428
Home Office     9343
Name: Segment, dtype: int64

In [66]:
newdf['Segment'] =  newdf['Segment'].map({'Home Office':0, 
                                        'Corporate':1,
                                        'Consumer':2})
newdf['Segment'].value_counts()

2    26518
1    15428
0     9343
Name: Segment, dtype: int64

In [67]:
print("Number of Unique COUNTRIES: ", len(newdf['Country'].unique()),"\nTOP 5\n", newdf['Country'].value_counts().head(5))
print("\nNumber of Unique STATES: ", len(newdf['State'].unique()),"\nTOP 5\n", newdf['State'].value_counts().head(5))
print("\nNumber of Unique CITIES: ", len(newdf['City'].unique()),"\nTOP 5\n", newdf['City'].value_counts().head(5))

Number of Unique COUNTRIES:  147 
TOP 5
 United States    9993
Australia        2837
France           2827
Mexico           2644
Germany          2065
Name: Country, dtype: int64

Number of Unique STATES:  1094 
TOP 5
 California       2001
England          1499
New York         1128
Texas             984
Ile-de-France     981
Name: State, dtype: int64

Number of Unique CITIES:  3636 
TOP 5
 New York City    915
Los Angeles      747
Philadelphia     537
San Francisco    510
Santo Domingo    443
Name: City, dtype: int64


In [68]:
newdf['Market'].value_counts()

APAC      11002
LATAM     10294
EU        10000
US         9993
EMEA       5029
Africa     4587
Canada      384
Name: Market, dtype: int64

In [69]:
newdf['Market'] =  newdf['Market'].map({'Canada':0, 
                                        'Africa':1,
                                        'EMEA':2,
                                        'US':3,
                                        'EU':4,
                                        'LATAM':5,
                                        'APAC':6})
newdf['Market'].value_counts()

6    11002
5    10294
4    10000
3     9993
2     5029
1     4587
0      384
Name: Market, dtype: int64

In [70]:
newdf['Region'].value_counts()

Central           11116
South              6645
EMEA               5029
North              4785
Africa             4587
Oceania            3487
West               3203
Southeast Asia     3129
East               2848
North Asia         2338
Central Asia       2048
Caribbean          1690
Canada              384
Name: Region, dtype: int64

In [71]:
newdf['Region'] =  newdf['Region'].map({'Canada':0, 
                                        'Caribbean':1,
                                        'Central Asia':2,
                                        'North Asia':3,
                                        'East':4,
                                        'Southeast Asia':5,
                                        'West':6,
                                        'Oceania':7,
                                        'Africa':8,
                                        'North':9,
                                        'EMEA':10,
                                        'South':11,
                                        'Central':12})
newdf['Region'].value_counts()

12    11116
11     6645
10     5029
9      4785
8      4587
7      3487
6      3203
5      3129
4      2848
3      2338
2      2048
1      1690
0       384
Name: Region, dtype: int64

In [72]:
newdf['Category'].value_counts()

Office Supplies    31272
Technology         10141
Furniture           9876
Name: Category, dtype: int64

In [73]:
newdf['Category'] = newdf['Category'].map({'Furniture':0, 
                                        'Technology':1,
                                        'Office Supplies':2})
newdf['Category'].value_counts()

2    31272
1    10141
0     9876
Name: Category, dtype: int64

In [74]:
newdf['Sub-Category'].value_counts()

Binders        6152
Storage        5059
Art            4883
Paper          3537
Chairs         3434
Phones         3357
Furnishings    3170
Accessories    3075
Labels         2606
Envelopes      2435
Supplies       2425
Fasteners      2420
Bookcases      2411
Copiers        2223
Appliances     1755
Machines       1486
Tables          861
Name: Sub-Category, dtype: int64

In [75]:
newdf['Sub-Category'] =  newdf['Sub-Category'].map({'Binders':0, 
                                        'Storage':1,
                                        'Art':2,
                                        'Paper':3,
                                        'Chairs':4,
                                        'Phones':5,
                                        'Furnishings':6,
                                        'Accessories':7,
                                        'Labels':8,
                                        'Envelopes':9,
                                        'Supplies':10,
                                        'Fasteners':11,
                                        'Bookcases':12,
                                        'Copiers':13,
                                        'Appliances':14,
                                        'Machines':15,
                                        'Tables':16})

In [76]:
newdf['Sub-Category'].value_counts()

0     6152
1     5059
2     4883
3     3537
4     3434
5     3357
6     3170
7     3075
8     2606
9     2435
10    2425
11    2420
12    2411
13    2223
14    1755
15    1486
16     861
Name: Sub-Category, dtype: int64

In [77]:
newdf['Order Priority'].value_counts()

Medium      29432
High        15501
Critical     3932
Low          2424
Name: Order Priority, dtype: int64

In [78]:
newdf['Order Priority'] =  newdf['Order Priority'].map({'Low':0, 
                                              'Critical':1,
                                              'High':2,
                                              'Medium':3})
newdf['Order Priority'].value_counts()

3    29432
2    15501
1     3932
0     2424
Name: Order Priority, dtype: int64

In [79]:
newdf.head()

Order Date   Ship Date  Ship Mode  Segment        City          State  \
0  31-12-2014  03-01-2015          2        1   S�o Paulo      S�o Paulo   
1  31-12-2014  01-01-2015          1        1     Jakarta        Jakarta   
2  31-12-2014  02-01-2015          2        2         Ufa  Bashkortostan   
3  31-12-2014  03-01-2015          1        1     Bangkok        Bangkok   
4  31-12-2014  05-01-2015          3        1  Maidenhead        England   

          Country  Market  Region  Category  Sub-Category  \
0          Brazil       5      11         1            13   
1       Indonesia       6       5         0            12   
2          Russia       2      10         1            13   
3        Thailand       6       5         0            16   
4  United Kingdom       4       9         1             5   

                              Product Name      Sales  Quantity  Discount  \
0                Canon Wireless Fax, Color  1264.4660         5     0.002   
1           Safco Library with Doors, Pine  1091.2806         3     0.070   
2           Hewlett Personal Copier, Laser   865.6200         6     0.000   
3  Lesro Training Table, Adjustable Height  1048.7313         9     0.570   
4                Motorola Audio Dock, VoIP   867.3000         5     0.000   

      Profit  Shipping Cost  Order Priority  
0   301.4660         253.25               2  
1    46.9206         243.11               2  
2    51.8400         138.18               2  
3 -1195.2387          86.86               2  
4   251.4000          53.16               3

DESCRIPTION OF THE DATAFRAME

In [80]:
newdf.describe()

Ship Mode       Segment        Market        Region      Category  \
count  51289.000000  51289.000000  51289.000000  51289.000000  51289.000000   
mean       2.348359      1.334867      3.940533      8.188929      1.417165   
std        0.914139      0.766205      1.602188      3.392634      0.792630   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        2.000000      1.000000      3.000000      6.000000      1.000000   
50%        3.000000      2.000000      4.000000      9.000000      2.000000   
75%        3.000000      2.000000      5.000000     11.000000      2.000000   
max        3.000000      2.000000      6.000000     12.000000      2.000000   

       Sub-Category         Sales      Quantity      Discount        Profit  \
count  51289.000000  51289.000000  51289.000000  51289.000000  51289.000000   
mean       6.016924    246.495075      3.476554      0.142906     28.611431   
std        4.681831    487.569052      2.278788      0.212282    174.342642   
min        0.000000      0.444000      1.000000      0.000000  -6599.978000   
25%        2.000000     30.766500      2.000000      0.000000      0.000000   
50%        5.000000     85.056000      3.000000      0.000000      9.240000   
75%       10.000000    251.067600      5.000000      0.200000     36.810000   
max       16.000000  22638.480000     14.000000      0.850000   8399.976000   

       Shipping Cost  Order Priority  
count   51289.000000    51289.000000  
mean       26.376403        2.402659  
std        57.297256        0.823064  
min         0.000000        0.000000  
25%         2.610000        2.000000  
50%         7.790000        3.000000  
75%        24.450000        3.000000  
max       933.570000        3.000000